### PR TITLE
feat(chips): Add keyboard navigation

### PR DESCRIPTION
**Pull request by patrickrodee**

### CODE REVIEW

File: packages/mdc-chips/chip-set/foundation.ts [hunk 1]
```diff
@@ -104,6 +111,183 @@ export class MDCChipSetFoundation extends MDCFoundation<MDCChipSetAdapter> {
...
+
+    // Early exit if the index is invalid or the source of the event
+    if (idx < 0 || idx > maxIndex || idx === index) {
+      return;
+    }
+
+    this.adapter_.focusChipAtIndex(idx);
+  }
+
+  private determineNextVerticalChip_(chipId: string, key: string): number {
```
Comment (abhiomkar) [inline packages/mdc-chips/chip-set/foundation.ts]: Can you point me to article (a11y) that explains if chips need to support vertical focus traversal?

IMO, UP arrow & LEFT arrow have the same behaviour - i.e., changes focus to previous chip. Similarly, DOWN & RIGHT arrow focuses the next chip in the list.

Comment (abhiomkar) [reply packages/mdc-chips/chip-set/foundation.ts]: As per this example in https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/LayoutGrids.html#

The only different that I understand between DOWN key and RIGHT key is that, DOWN moves focus to next logical row (A chip containing content and remove icon) where as RIGHT moves focus to every element.

WDYT?

File: packages/mdc-chips/chip-set/foundation.ts [hunk 2]
```diff
@@ -104,6 +111,183 @@ export class MDCChipSetFoundation extends MDCFoundation<MDCChipSetAdapter> {
     this.adapter_.removeChip(chipId);
   }
 
+  handleChipKeyboard(chipId: string, key: string) {
```
[the remark this answers is not in the document]
Comment (patrickrodee) [reply packages/mdc-chips/chip-set/foundation.ts]: Done

File: packages/mdc-chips/chip-set/foundation.ts [hunk 3]
```diff
@@ -104,6 +111,183 @@ export class MDCChipSetFoundation extends MDCFoundation<MDCChipSetAdapter> {
...
+    const maxIndex = this.adapter_.getChipListLength() - 1;
```
[the remark this answers is not in the document]
Comment (patrickrodee) [reply packages/mdc-chips/chip-set/foundation.ts]: Done

File: packages/mdc-chips/chip-set/foundation.ts (hunk 1 above)
Comment (patrickrodee) [reply packages/mdc-chips/chip-set/foundation.ts]: Replied on Slack

**Comment by stevenyxu** [comment]
This PR was closed. Was this change abandoned or incorporated elsewhere? @VanessaNSchmitt

**Pull request closed without merging**